components/layouts/sidebar.tsx

```diff
@@ -1,5 +1,77 @@
+'use client';
+
+import { BookIcon, HomeIcon, SettingIcon } from '@/assets/icons';
+import { Logo } from '@/assets/svg-images';
+import Image from 'next/image';
+import { useRouter } from 'next/navigation';
+import { useState } from 'react';
+
+const sidebarItem = [
+  {
+    id: 1,
+    iconName: 'Home',
+    path: '/dashboard',
+    icon: <HomeIcon />,
+    position: -15,
+  },
+  {
+    id: 2,
+    iconName: 'Book',
+    path: '/leads',
+    icon: <BookIcon />,
+    position: 50,
+  },
+  {
+    id: 3,
+    iconName: 'Setting',
+    path: '/settings',
+    icon: <SettingIcon />,
+    position: 110,
+  },
+];
+
 const Sidebar = () => {
-  return <div className=''>Sidebar</div>;
+  const [selected, setSelected] = useState(-15);
+  const router = useRouter();
+
+  const handleClick = (position: number, path: string) => {
+    setSelected(position);
+    router.push(path);
+  };
+
+  return (
+    <div className='w-[88px] bg-white relative'>
+      <div className='absolute w-full flex justify-center top-5'>
+        <Logo />
+      </div>
+
+      <div className='relative left-1/2 top-1/2 transform -translate-x-1/2 -translate-y-1/2 flex flex-col h-[147px] justify-between items-center w-full'>
+        {sidebarItem.map((item) => {
+          return (
```

Comment:
Here we can skip explicit return into JSX. We will return implicitly like:
```html
{sidebarItem.map((item) => (
  <p
    key={item?.id}
    onClick={() => handleClick(item?.position, item?.path)}
    className={`cursor-pointer transition-all duration-500 ${
      selected === item?.position ? 'text-[#5630FF]' : 'text-[#69708C]'
    }`}>
      {item?.icon}
  </p>
)}
```

Also we can destructure the `position` and `path` properties from item on each iteration. Thus our code will looks like:
```html
{sidebarItem.map(({ position, path }) => (
  <p
    key={item?.id}
    onClick={() => handleClick(position, path)}
    className={`cursor-pointer transition-all duration-500 ${
      selected === item?.position ? 'text-[#5630FF]' : 'text-[#69708C]'
    }`}>
      {item?.icon}
  </p>
)}
```